tap_exacttarget/dao.py

```diff
@@ -36,9 +38,31 @@ def load_schema(stream):
 
     return schema
 
+# boolean function to check if the error is 'timeout' error or not
+def is_timeout_error():
+    """
+        This function checks whether the URLError contains 'timed out' substring and return boolean
+        values accordingly, to decide whether to backoff or not.
+    """
+    def gen_fn(exc):
+        if str(exc).__contains__('timed out'):
+            # retry if the error string contains 'timed out'
+            return False
+        return True
+
+    return gen_fn
+
 # decorator for retrying on error
 def exacttarget_error_handling(fnc):
-    @backoff.on_exception(backoff.expo, (socket.timeout, ConnectionError), max_tries=5, factor=2)
+    @backoff.on_exception(backoff.expo,
+                          urllib.error.URLError, # backoff 'timeout' error for SOAP API
+                          giveup=is_timeout_error(),
+                          max_tries=5,
+                          factor=2)
+    @backoff.on_exception(backoff.expo,
+                          (socket.timeout, ConnectionError, requests.Timeout),
```

Comment:
Do we need request.Timeout, socket.timeout, and URlError?

Reply:
Where is the import for ConnectionError?

Reply:
The socket.timeout was added as part of TDL-14621: Add retry logic to requests and TDL-14622: Retry ConnectionResetErrors.

This ConnectionError is the error regarding the backoff we added for ConnectionResetErrors